src/customHooks/useProducts.tsx

```diff
@@ -0,0 +1,22 @@
+import { useState } from "react";
+import { ProductType } from '../types';
+
+export default (initialValue: ProductType[]) => {
+  const [ products, setProducts ] = useState<ProductType[]>(initialValue);
```

Comment:
I love that you created this custom hook, but I think you could have added your data fetch into here as well instead of putting that logic into App.tsx. I think that would have made it a bit more beneficial + clear.